clients/vscode/src/chat/ChatViewProvider.ts

```diff
@@ -508,3 +511,36 @@ function resolveFilePathAndGitUrl(uri: Uri, gitProvider: GitProvider): { filepat
     git_url: remoteUrl ?? "",
   };
 }
+
+async function resolveDocument(
+  logger: LogOutputChannel,
+  folders: readonly WorkspaceFolder[] | undefined,
+  filepath: string,
+): Promise<TextDocument | null> {
+  if (filepath.startsWith("file://")) {
+    const absoluteFilepath = Uri.parse(filepath, true);
+    return workspace.openTextDocument(absoluteFilepath);
+  }
+
+  if (!folders) {
+    return null;
+  }
+
+  for (const root of folders) {
+    const absoluteFilepath = Uri.joinPath(root.uri, filepath);
+    try {
+      return workspace.openTextDocument(absoluteFilepath);
```

Comment:
```suggestion
      return await workspace.openTextDocument(absoluteFilepath);
```
await is required for catching errors.